### PR TITLE
Enable WAL by default in single process config

### PR DESCRIPTION
**Pull request by codesome**
Signed-off-by: Ganesh Vernekar <cs15btech11018@iith.ac.in>

**What this PR does**:

**Which issue(s) this PR fixes**:
Fixes #<issue number>

**Checklist**
- [ ] Tests updated
- [ ] Documentation added
- [ ] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`

### CODE REVIEW

File: docs/configuration/single-process-config.yaml [hunk 1]
```diff
@@ -33,6 +33,11 @@ ingester:
   spread_flushes: true
   chunk_age_jitter: 0
 
+  walconfig:
+    wal_enabled: true
+    recover_from_wal: true
+    wal_dir: /tmp/cortex/wal
```
Comment (pracucci) [inline docs/configuration/single-process-config.yaml:39]: Running Cortex with the default image fails with the following error:
`failed to recover from WAL: open /tmp/cortex/wal: no such file or directory`

This is not just an error you get when running integration tests but also running it locally with (the following command picked from the doc):
`./cortex -config.file=./docs/configuration/single-process-config.yaml`

Comment (codesome) [reply docs/configuration/single-process-config.yaml:39]: Yes, I found that out before seeing this review and have already pushed a fix for it. Now a directory is created when WAL or recovery from WAL is enabled.

Comment (codesome) [reply docs/configuration/single-process-config.yaml:39]: Integration tests are passing now :)

File: pkg/ingester/ingester.go [hunk 2]
```diff
@@ -157,6 +158,12 @@ func New(cfg Config, clientConfig client.Config, limits *validation.Overrides, c
 		}
 	}
 
+	if cfg.WALConfig.WALEnabled || cfg.WALConfig.Recover {
+		if err := os.MkdirAll(cfg.WALConfig.Dir, 0777); err != nil {
```
Comment (pracucci) [inline pkg/ingester/ingester.go:162]:
```suggestion
		if err := os.MkdirAll(cfg.WALConfig.Dir, os.ModePerm); err != nil {
```

**Pull request merged**